tests/unit/amo/components/TestConfirmButton.js

```diff
@@ -27,138 +22,147 @@ describe(__filename, () => {
+    expect(screen.getByText('some warning message')).toBeInTheDocument();
+    expect(screen.getByClassName('ConfirmationDialog')).toBeInTheDocument();
+    expect(
+      screen.getByClassName('ConfirmButton--show-confirmation'),
+    ).toBeInTheDocument();
   });
 
   it('configures ConfirmationDialog', () => {
     const cancelButtonText = 'Nevermind, take me back';
-    const cancelButtonType = 'alert';
     const confirmButtonText = 'Do it!';
-    const confirmButtonType = 'alert';
     const message = 'Do you really want to cancel?';
-    const puffyButtons = true;
 
-    const root = renderWithDialog({
+    renderWithDialog({
       cancelButtonText,
-      cancelButtonType,
+      cancelButtonType: 'alert',
       confirmButtonText,
-      confirmButtonType,
+      confirmButtonType: 'alert',
       message,
-      puffyButtons,
+      puffyButtons: true,
     });
 
-    const confirmDialog = root.find(ConfirmationDialog);
-    expect(confirmDialog).toHaveProp('cancelButtonText', cancelButtonText);
-    expect(confirmDialog).toHaveProp('cancelButtonType', cancelButtonType);
-    expect(confirmDialog).toHaveProp('confirmButtonText', confirmButtonText);
-    expect(confirmDialog).toHaveProp('confirmButtonType', confirmButtonType);
-    expect(confirmDialog).toHaveProp('message', message);
-    expect(confirmDialog).toHaveProp('puffyButtons', puffyButtons);
+    expect(screen.getByText(message)).toBeInTheDocument();
+
+    const dialogCancelButton = screen.getByRole('button', {
+      name: cancelButtonText,
+    });
+    expect(dialogCancelButton).toHaveClass('ConfirmationDialog-cancel-button');
+    expect(dialogCancelButton).toHaveClass('Button--alert');
+    expect(dialogCancelButton).toHaveClass('Button--puffy');
+
+    const dialogConfirmButton = screen.getByRole('button', {
+      name: confirmButtonText,
+    });
+    expect(dialogConfirmButton).toHaveClass(
+      'ConfirmationDialog-confirm-button',
+    );
+    expect(dialogConfirmButton).toHaveClass('Button--alert');
+    expect(dialogConfirmButton).toHaveClass('Button--puffy');
   });
 
   it('hides the default button after it is clicked', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ store });
-
-    root
-      .find('.ConfirmButton-default-button')
-      .simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
+    render();
 
-    expect(root.find('.ConfirmButton-default-button')).toHaveLength(0);
+    const button = screen.getByRole('button', {
+      name: 'the default text of this button',
+    });
+    userEvent.click(button);
+    expect(button).not.toBeInTheDocument();
   });
 
-  it('hides ConfirmationDialog via onCancel', () => {
-    const { store } = dispatchClientMetadata();
-    const root = renderWithDialog({ store });
+  it('handles onCancel callback and hides ConfirmationDialog on cancel', () => {
+    const onCancel = jest.fn();
+    renderWithDialog({ onCancel });
+
+    expect(
```

Comment:
We've already asserted about this in `shows ConfirmationDialog when button is clicked`, and `renderWithDialog` also includes an assertion which guarantees that the `ConfirmationDialog` is visible.